web/src/lib/components/pages/record/profile-actions.tsx

```diff
@@ -2,55 +2,69 @@ import { h, Component } from 'preact';
 import Profile from '../profile/profile';
 import User from '../../../user';
 
-interface WhyProfileProps {
-  onClose: () => any;
+interface WhyProfileProps {}
```

Comment:
minor: since Props are no longer needed for this, you can specify `extends Component<void, WhyProfileState>` instead of using an empty interface :)

other than that: LGTM

Reply:
While I would like that better as well, that yields the following error for me:

`Type '{}' is not assignable to type 'void & ComponentProps<WhyProfile>'.
  Type '{}' is not assignable to type 'void'.`

Reply:
aaah, works with `class WhyProfile extends Component<{}, WhyProfileState>`

Reply:
whaaat. that's weird D:

i'll try to look into this (void worked for me in a similar situation)

Reply:
okay, for my own curiosity and sanity, i looked in to this.

the problem occurs at the usage site of `<WhyProfile />`, because the JSX transform invokes the component creator with an empty object in this case. For `T extends Component<P, S>`, this props object must be of type `P & ComponentProps<T>` (an [intersection type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types)). if `P === void`, the resulting `void & ComponentProps === void` (nothing is an instance of `void`, so nothing can be an instance of an even more specific type).

Besides using the empty object type `{}`, as you figured out, using [the types for `null` or `undefined`](https://www.typescriptlang.org/docs/handbook/basic-types.html#null-and-undefined) would also work, since all three of them are neutral in regards to type intersection. (that's actually kinda weird, but has practical no bearing on our situation).

Curiosity satisfied!

Reply:
I marvel at your curiosity. Thanks for the insight!